modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,326 @@
+            List<String> annotatedMethods
+        ) {
+            this.annotatedClasses = annotatedClasses;
+            this.annotatedMethods = annotatedMethods;
+        }
+
+        public List<String> getAnnotatedClasses() {
+            return annotatedClasses;
+        }
+
+        public List<String> getAnnotatedMethods() {
+            return annotatedMethods;
+        }
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations,
+        RoundEnvironment roundEnv
+    ) {
+        for (TypeElement annotation : annotations) {
+            AnnotatedElements annotatedElements =
+                getAnnotatedElements(roundEnv, annotation);
+            if (!(annotatedElements.getAnnotatedClasses().isEmpty() &&
+                annotatedElements.getAnnotatedMethods().isEmpty())
+            ) {
+                File outputDir = getOutputDirectory();
+                outputDir.mkdirs();
+                makeFile(new File(outputDir,
+                    ModernizerAnnotationUtils.IGNORE_CLASSES_FILE_NAME),
+                    annotatedElements.getAnnotatedClasses());
+                makeFile(new File(outputDir,
+                    ModernizerAnnotationUtils.IGNORE_METHODS_FILE_NAME),
+                    annotatedElements.getAnnotatedMethods());
+            }
+        }
+        return true;
+    }
+
+    private AnnotatedElements getAnnotatedElements(
+        RoundEnvironment roundEnv,
+        TypeElement annotation
+    ) {
+        List<String> annotatedClasses = new ArrayList<String>();
+        List<String> annotatedMethods = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(
+                        getFullClassNameRegex(element));
+                } else if (element.getKind().equals(ElementKind.METHOD) ||
+                    element.getKind().equals(ElementKind.CONSTRUCTOR)) {
+                    annotatedMethods.add(getMethodRepresentation(element));
+                }
+            }
+        }
+        return new AnnotatedElements(annotatedClasses, annotatedMethods);
+    }
+
+
+    private void makeFile(
+        File file,
+        List<String> annotatedElements
+    ) {
+        if (annotatedElements.isEmpty()) {
+            return;
+        }
+        Writer writer = null;
+        try {
+            writer = new BufferedWriter(new FileWriter(file));
+            for (String element : annotatedElements) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
```

Comment:
Call `Utils.closeQuietly` instead.

Reply:
`Utils` is in the maven-plugin module, whereas this code is in the annotation-processor module. Do you want me to move it?